### PR TITLE
New RSS Update

### PR DESCRIPTION
**Pull request by YuanbinLiu**
This PR is mainly about adding RSS new features and modifying function names in workflow.

### CODE REVIEW

File: autoplex/data/common/jobs.py [hunk 1]
```diff
@@ -611,14 +641,21 @@ def VASP_collect_data(
     return {"vasp_ref_dir": vasp_ref_dir, "isol_es": isol_es}
 
 
-def check_convergence_vasp(file):
+def check_convergence_vasp(path: str) -> bool:
```
Comment (JaGeo) [inline autoplex/data/common/jobs.py:644]: In the long run, we should rely on custodian handlers again. This is something that can be checked within custodian.

File: autoplex/data/common/jobs.py [hunk 2]
```diff
@@ -648,24 +685,65 @@ def safe_strip_hostname(value):
...
     atoms = (
-        data_distillation(vasp_ref_dir, f_max, force_label)
+        data_distillation(vasp_ref_dir, force_max, force_label)
         if distillation
         else read(vasp_ref_dir, index=":")
     )
 
-    train_structures, test_structures = stratified_dataset_split(atoms, test_ratio)
+    if test_ratio == 0 or test_ratio is None:
+        train_structures, test_structures = atoms, atoms
```
Comment (JaGeo) [inline autoplex/data/common/jobs.py:744]: Okay, great. This solves one of my issues as well. Thus, we train, test on the same data if not enough data points are there for a stratified split.

**Comment by JaGeo** [comment]
@YuanbinLiu Should this one be closed or does it contain updates needed for the RSS functionality?

**Comment by YuanbinLiu** [comment]
> > Hi @YuanbinLiu ,
> > I made a first round through the code. I think it overall looks very good.
> > I think the following parts would be very important for the code as well:
> > 
> > * [x]   one automatic flow performing the RSS procedure. It should ideally live in `auto/rss`
> > * [ ]   a tutorial for usage of the RSS part. Ideally, more information on the fitting part would also be included in the documentation. I think some of the settings there are still quite hard to grasp for a first time user of the code
> > * [x]   all tests would need to pass
> 
> @JaGeo Great suggestions. I am now handling them one by one.

The tutorial is being prepared. I will push a new PR for this.

File: autoplex/data/common/utils.py [hunk 3]
```diff
@@ -1051,17 +1054,17 @@ def rmv(v):
...
     Parameters
     ----------
-    e : float
+    e: float
         The energy for which to calculate the Boltzmann factor.
-    emin : float
+    emin: float
         The minimum energy to consider in the calculation.
-    kT : float
+    kt: float
```
[the remark this answers is not in the document]
Comment (YuanbinLiu) [reply autoplex/data/common/utils.py]: Here I believe that kt should work, as it is more commonly used in thermodynamics and is easier to recognize.

File: autoplex/data/common/jobs.py (hunk 1 above)
Comment (YuanbinLiu) [reply autoplex/data/common/jobs.py:644]: We've implemented the custodian handlers now, so I'll go ahead and remove the function.

**Comment by JaGeo** [comment]
> > > Hi @YuanbinLiu ,
> > > I made a first round through the code. I think it overall looks very good.
> > > I think the following parts would be very important for the code as well:
> > > 
> > > * [x]    one automatic flow performing the RSS procedure. It should ideally live in `auto/rss`
> > > * [ ]    a tutorial for usage of the RSS part. Ideally, more information on the fitting part would also be included in the documentation. I think some of the settings there are still quite hard to grasp for a first time user of the code
> > > * [x]    all tests would need to pass
> > 
> > 
> > @JaGeo Great suggestions. I am now handling them one by one.
> 
> The tutorial is being prepared. I will open a new PR for this.

Yes, please go ahead with a new PR.

File: autoplex/data/rss/jobs.py [hunk 4]
```diff
@@ -171,23 +240,41 @@ def _update_buildcell_options(self, updates, origin):
...
+                flag = not (flag and i.startswith("%"))
+                contents.append(i + "\n")
+            else:
+                contents.append("#" + i + "\n")
 
         with open(bc_file, "w") as f:
             f.writelines(contents)
 
-    def _is_metal(self, element_symbol):
+    def _is_metal(self, element_symbol: str) -> bool:
```
[the remark this answers is not in the document]
Comment (YuanbinLiu) [reply autoplex/data/rss/jobs.py:284]: This has now been implemented.

**Comment by YuanbinLiu** [comment]
> @YuanbinLiu Should this one be closed or does it contain updates needed for the RSS functionality?

@JaGeo all comments in this PR have been addressed. Please close it.

**Comment by JaGeo** [comment]
@YuanbinLiu I will close it and I will go through the other pull request tomorrow then.

**Pull request closed without merging**